x-pack/legacy/plugins/task_manager/task.ts

```diff
@@ -10,6 +10,8 @@ import Joi from 'joi';
  * Type definitions and validations for tasks.
  */
 
+type Require<T extends object, P extends keyof T = keyof T> = Omit<T, P> & Required<Pick<T, P>>;
```

Comment:
I think this new `Require` type could use a short comment - not immediately clear what it's doing :-)